src/librustc_driver/driver.rs

```diff
@@ -1147,15 +1139,14 @@ pub fn phase_4_translate_to_llvm<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 
 /// Run LLVM itself, producing a bitcode file, assembly file or object file
 /// as a side effect.
-#[cfg(feature="llvm")]
-pub fn phase_5_run_llvm_passes(sess: &Session,
+pub fn phase_5_run_llvm_passes<T: TransCrate>(sess: &Session,
```

Comment:
nit: style is to give "significant" type parameters a significant name, e.g. call them `TRANS` instead of `T`.